scripts/docs/python_ref_builder.py

```diff
@@ -0,0 +1,155 @@
+# (C) 2023 GoodData Corporation
+import argparse
+import json
+from pathlib import Path
+from typing import List
+
+MODULE_TEMPLATE_STRING = Path("module_template.md").read_text()
+CLASS_TEMPLATE_STRING = Path("class_template.md").read_text()
+FUNCTION_TEMPLATE_STRING = Path("function_template.md").read_text()
+
+
+def process_json_file(file_path: str) -> dict:
+    """Load JSON data from a file.
+
+    Args:
+        file_path (str): Path to the JSON file.
+
+    Returns:
+        dict: Parsed JSON data.
+    """
+    with open(file_path) as json_file:
+        return json.load(json_file)
+
+
+def create_file_structure(data: dict, root: Path, url_root: str):
+    """Recursively create file structure based on JSON data.
+
+    Args:
+        data (dict): JSON data representing the object.
+        root (Path): Path to the root directory.
+        url_root (str): URL root path for the API reference.
+    """
+    links = {}
+
+    def _recursive_create(data_root: dict, dir_root: Path, url_root: str, module_import_path: str):
+        """Recursively create files and directories.
+
+        Args:
+            data_root (dict): Sub-dictionary of the JSON representing the object.
+            dir_root (Path): Path to the directory root.
+            url_root (str): URL root path for the API reference.
+            module_import_path (str): Import path to the object.
+        """
+        dir_root.mkdir(exist_ok=True)
+        for name, obj in data_root.items():
+            # There are entries in the json, that are not dicts (ex: the field `kind`)
+            if not isinstance(obj, dict):
+                continue
+
+            # If an object already has a page, skip it
+            if name in links:
+                continue
+
+            kind = obj.get("kind", None)
+
+            obj_module_import_path = module_import_path + f".{name}" if module_import_path != "" else name
+
+            # Remove ".functions" from the path, to correspond to the import path
+            if ".functions" in obj_module_import_path:
+                obj_module_import_path = obj_module_import_path.replace(".functions", "")
+
+            # Create files based on the kind of the data: module/class/function
+            if kind == "module":
+                (dir_root / name).mkdir(exist_ok=True)
+                (dir_root / name / "_index.md").open("w+").write(
+                    MODULE_TEMPLATE_STRING.replace("PATH", obj_module_import_path)
+                    .replace("NAME", name)
+                    .replace("LINK", name)
+                )
+                links[name] = {"path": f"{url_root}/{name}".lower(), "kind": "function"}  # Lowercase for Hugo
```

Comment:
It seems that some code is repeating. Create utility function that does the majority of the repeating code.